src/cockpit/channels/dbus.py

```diff
@@ -205,17 +244,44 @@ def do_open(self, options):
             if err.errno != errno.EBUSY:
                 raise
 
-        self.ready()
-
-    def add_match(self, rule, handler):
+        if self.name is not None:
+            async def get_ready():
+                async with self.watch_processing_lock:
+                    await self.setup_name_owner_tracking()
+                    if self.owner:
+                        self.ready(unique_name=self.owner)
+                    else:
+                        self.close(problem="not-found")
+            task = asyncio.create_task(get_ready())
+            self.tasks.add(task)
+            task.add_done_callback(self.tasks.discard)
+        else:
+            self.ready()
+
+    def add_signal_handler(self, rule, handler):
```

Comment:
I think I'd prefer to see this as a function which takes the handler as the first argument and the match rule as `**kwargs`.

Reply:
Done.